src/main/java/seedu/address/logic/commands/recipe/RecipeAddStepCommand.java

```diff
@@ -47,17 +48,24 @@ public CommandResult execute(Model model) throws CommandException {
 
         Recipe recipeToEdit = lastShownList.get(recipeIndex.getZeroBased());
         UniqueStepList targetStepList = recipeToEdit.getSteps();
-
-        if (stepIndex.getZeroBased() > targetStepList.asUnmodifiableObservableList().size()) {
-            // ensure the step index is valid
-            throw new CommandException((Messages.MESSAGE_INVALID_STEP_DISPLAYED_INDEX));
-        }
+        Index newStepIndex;
 
         if (targetStepList.contains(toAdd)) {
             throw new CommandException(((Messages.MESSAGE_DUPLICATE_STEPS)));
         }
 
-        targetStepList.add(stepIndex, toAdd);
+        if (stepIndex.isPresent()) {
+            if (stepIndex.get().getZeroBased() > targetStepList.asUnmodifiableObservableList().size()) {
+                // ensure the step index is valid
+                throw new CommandException(String.format(MESSAGE_INVALID_STEP_DISPLAYED_INDEX,
+                        RecipeAddCommand.MESSAGE_USAGE));
+            }
+            targetStepList.add(stepIndex.get(), toAdd);
+        } else {
+            newStepIndex = new Index(targetStepList
+                    .asUnmodifiableObservableList().size());
+            targetStepList.add(newStepIndex, toAdd);
```

Comment:
Maybe we can have an add method for `UniqueStepList` that adds to the end of the list by default, if no index is provided, like the `add()` method of `java.util.List`. Then we can just call that method here directly.

Reply:
okay sound good i will implement in now

Reply:
eh wait its already implemented haha